MB-14433-Removed prop injection from page components and deleted the base path prop.

dymonmoore opened the pull request:
## [Jira ticket](https://dp3.atlassian.net/browse/MB-14487) for this change

## Summary

This pull request is an incremental change apart of the React Admin upgrade. Prop injection was removed from all React Admin Page components (i.e List, Create, Edit) and the basePath prop was removed from the OfficeUserList component. 

The app will not compile until the rest of the upgrade changes are complete. This pull request cannot be tested at this time. For now, all is needed is a **code review.**

Current State of Admin App without Prop Injection due to upgrade being incomplete.
<img width="995" alt="Screen Shot 2022-11-29 at 10 34 25 AM" src="https://user-images.githubusercontent.com/111006369/204572981-637df876-e901-41aa-b8a6-4a735b277498.png">

robot-mymove commented:
<!--
  1 failure:  Yarn Audit Issues...
  2 warning:  This PR does not ..., Please add the JI...
  
  
  DangerID: danger-id-Danger;
-->

<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Yarn Audit Issues Found:
critical - Prototype pollution in webpack loader-utils
Package loader-utils
Patched in >=1.4.1
Dependency of @trussworks/react-file-viewer
Path @trussworks/react-file-viewer > pdfjs-dist > worker-loader > loader-utils
More info https://github.com/advisories/GHSA-76p3-8jx3-jpfq

high - loader-utils is vulnerable to Regular Expression Denial of Service (ReDoS) via url variable
Package loader-utils
Patched in >=1.4.2
Dependency of @trussworks/react-file-viewer
Path @trussworks/react-file-viewer > pdfjs-dist > worker-loader > loader-utils
More info https://github.com/advisories/GHSA-3rfm-jhwj-7488

high - loader-utils is vulnerable to Regular Expression Denial of Service (ReDoS)
Package loader-utils
Patched in >=1.4.2
Dependency of @trussworks/react-file-viewer
Path @trussworks/react-file-viewer > pdfjs-dist > worker-loader > loader-utils
More info https://github.com/advisories/GHSA-hhq3-ff78-jv3g

2 high vulnerabilities and 1 critical vulnerabilities found</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This PR does not include changes to unit tests, even though it affects app code.</td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>Please add the JIRA issue key to the PR title (e.g. MB-123)</td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 19b6f16740b3898c79b6f9d82b53502d413a8684
</p>

dymonmoore commented:
Hi team! I removed prop injection from the React Admin page components and deleted the use of baseProp according to the upgrade guide. There is some additional prop injection in non-page components in the admin app. Should it be removed? Here's examples:  

const OfficeUserEditToolbar = (props) => (
  <Toolbar {...props}>
    <SaveButton />
  </Toolbar>
);

const OfficeUserListFilter = (props) => (
  <Filter {...props}>
    <TextInput source="search" alwaysOn />
  </Filter>
);

PDickeyTruss reviewed (approved):
Looks great, approved! 

Re your question about if you should remove the additional prop injection in non-page components in the admin app. I think you were correct to leave those as is, I see them mention 'page components' in that area of the guide. 👍
